presto-main/src/main/java/com/facebook/presto/operator/ChannelSet.java

```diff
@@ -100,16 +108,34 @@ public int size()
             return hash.getGroupCount();
         }
 
-        public void addPage(Page page)
+        /**
+         * This function adds a page to a work object for later processing.
+         * @param page A data page whose data needs to be added to the hash table.
+         * @return The work needs to be done on this page.
+         */
+        public Work<?> addPage(Page page)
         {
             Work<?> work = hash.addPage(page);
-            boolean done = work.process();
-            // TODO: this class does not yield wrt memory limit; enable it
-            verify(done);
+            return work;
+        }
+
+        /**
+         * The call back function to reserve memory.
+         * @return true if the memory reservation was successful.
+         */
+        public boolean updateMemoryReservation()
+        {
+            // If memory is not available, once we return, this operator will be blocked until memory is available.
+            operatorContext.setMemoryReservation(getEstimatedSize());
```

Comment:
In `addPage(Page)` `operatorContext` was considered nullable. (don't know why). Please ensure it wasn't necessary.
(If it wasn't, it'd be better if removing null check was a separate, preparatory commit)

Reply:
I don't think it was necessary because SetBuilderOperator.operatorContext was requireNonNull since creation. But the purpose of removing line 110 and 11 was not to remove the not null check, but removing the setMemoryReservation from addPage(Page page).